publisher-sdk/src/main/java/com/criteo/publisher/CriteoBannerView.java

```diff
@@ -41,14 +44,39 @@
    * having to init the SDK before.
    */
   @Nullable
-  private final Criteo criteo;
+  private Criteo criteo;
 
   @Nullable
   private CriteoBannerAdListener criteoBannerAdListener;
 
   @Nullable
   private CriteoBannerEventController criteoBannerEventController;
 
+  /**
+   * Used when setting {@link CriteoBannerView} in XML
+   */
+  public CriteoBannerView(@NonNull Context context, AttributeSet attrs) {
+    super(context, attrs);
+    TypedArray a = context.getTheme().obtainStyledAttributes(
+        attrs,
+        R.styleable.CriteoBannerView,
+        0,
+        0
+    );
+
+    try {
+      int width = a.getInteger(R.styleable.CriteoBannerView_criteoAdUnitWidth, -1);
+      int height = a.getInteger(R.styleable.CriteoBannerView_criteoAdUnitHeight, -1);
+      String adUnitId = a.getString(R.styleable.CriteoBannerView_criteoAdUnitId);
+
+      if (adUnitId != null && width >= 0 && height >= 0) {
```

Comment:
What do you think about being strict on the input: when at least one field is given, then standalone is targetted and an exception is thrown if: `adUnitId == null || width < 0 || height < 0`. When no field is given, then InHouse is targetted and no check is done.

Reply:
I think it makes the output uncertain and adds unneeded complexity. For InHouse the field that will be set via `XML` will not even be taken into account because `loadBid` takes an `AdUnit` as a param which will be used instead.

Reply:
For InHouse there is no issue. But for Standalone, a publisher setting a wrong setting will get its `BannerAdUnit` nullified silently and would only get no bid. Throwing early here would make it clear that input is incorrect.

Reply:
Yet there is no guarantee that the publisher is using `InHouse` if all three fields are not present.  I would find it confusing to throw only if (!a || !b || !c) and not if (!a && !b && !c). I would rather have a good log that informs the publisher about the initialization status than throw here.

Reply:
Something like:

```
        logger.warn(
            "The AdUnit associated to this CriteoBannerView instance was not initialized as: "
                + "criteoAdUnitId, criteoAdUnitHeight and criteoAdUnitWidth were not all set. If you are" 
                + "using CriteoBannerView as part of the InHouse integration then this is normal. " 
                + "Otherwise, make sure to set all three attributes."
        );
```

Reply:
Ok for having a log to indicate the expected integration. However, I still think that the intermediate state is an error.
Isn't it possible to have:
- all correctly set -> `logger.info("CriteoBannerView inflated for Standalone integration.")`
- no setting -> `logger.info("CriteoBannerView inflated for InHouse integration.")`
- missing or incorrect -> throw explaining that InHouse expect nothing and Standalone expect all.

Note that I'm using a log INFO here: InHouse is the second most used integration. We shouldn't warn so often for a normal case.

Reply:
Alright, that's a good suggestion. Another option would be to have dedicated views per integration: `InHouseCriteoBannerView` and `StandaloneCriteoBannerView`. That would make these widgets more cohesive. Naming could be fine-tuned. WDYT?

Reply:
Indeed it would make it simpler at implementation level.

OTOH, we would end with `integration * format` different objects. Integration is `{Standalone, InHouse, ServerBidding}` and format is `{Banner, Interstitial, Native}`. That would already make 9 objects almost identical :( For sure it would be possible to use `abstract` classes to factorize the mess but it would still make 9 objects so that's not simpler at SDK level.

I'm not fan of what we have currently. What you're proposing is a lot of work, and I want to be sure that it is worth it.

Reply:
Ok `lot of work` was my intuition as well. Let's keep it simple then.